src/s3ql/statfs.py

```diff
@@ -49,32 +55,89 @@ def main(args=None):
         pprint = pretty_print_size
 
     ctrlfile = assert_fs_owner(options.mountpoint, mountpoint=True)
-
-    # Use a decent sized buffer, otherwise the statistics have to be
-    # calculated three(!) times because we need to invoke getxattr
-    # three times.
-    buf = llfuse.getxattr(ctrlfile, 's3qlstat', size_guess=256)
-
-    (entries, blocks, inodes, fs_size, dedup_size,
-     compr_size, db_size, cache_cnt, cache_size, dirty_cnt,
-     dirty_size, removal_cnt) = struct.unpack('QQQQQQQQQQQQ', buf)
-    p_dedup = dedup_size * 100 / fs_size if fs_size else 0
-    p_compr_1 = compr_size * 100 / fs_size if fs_size else 0
-    p_compr_2 = compr_size * 100 / dedup_size if dedup_size else 0
-    print ('Directory entries:    %d' % entries,
-           'Inodes:               %d' % inodes,
-           'Data blocks:          %d' % blocks,
-           'Total data size:      %s' % pprint(fs_size),
-           'After de-duplication: %s (%.2f%% of total)'
-             % (pprint(dedup_size), p_dedup),
-           'After compression:    %s (%.2f%% of total, %.2f%% of de-duplicated)'
-             % (pprint(compr_size), p_compr_1, p_compr_2),
-           'Database size:        %s (uncompressed)' % pprint(db_size),
-           'Cache size:           %s, %d entries' % (pprint(cache_size), cache_cnt),
-           'Cache size (dirty):   %s, %d entries' % (pprint(dirty_size), dirty_cnt),
-           'Queued object removals: %d' % (removal_cnt,),
-           sep='\n')
+    if options.prometheus_exporter:        
```

Comment:
I don't think that's a good idea. You should start the promethous exporter in a separate thread in mount(), and leave statfs as-is.

Reply:
Yes you are right I tried but I am not an expert In python I getted the FS fully blocked, maybe you could help me with that.

Reply:
Try to model your implementation after the `CommitThread` in `mount.py`, that should work fine. I'll help further in the other pull request.